### ISSUE TITLE
Reported results for RNNModel and DistillerRNNModel are not equal

### ISSUE THREAD

**Issue by hadifar** [NervanaSystems/distiller#383]
Hey folks,
First of all thanks for your awesome work!
I tried to run the [word_language_model](https://github.com/NervanaSystems/distiller/tree/master/examples/word_language_model) example but it seems in the [model.py](https://github.com/NervanaSystems/distiller/blob/master/examples/word_language_model/model.py) two implementations of RNN exists. `DistillerRNNModel` and `RNNModel` which the first one is your own LSTM version cannot reach the 7x/8x perplexity that reported in the readme. I replaced the following line in [main.py](https://github.com/NervanaSystems/distiller/blob/master/examples/word_language_model/main.py)

`model.RNNModel(ntokens, args.emsize, args.nhid, args.nlayers, args.dropout, args.tied)`

with:

`model = model.DistillerRNNModel(ntokens, args.emsize, args.nhid, args.nlayers, args.dropout, args.tied)
`

Then I run the following command:

`python3 main.py --cuda --emsize 1500 --nhid 1500 --dropout 0.65 --tied --wd=1e-6`

I would like to know do I miss something or its a problem with the new version of pytorch? 
Note that the RNNModel (pytorch LSTM implementation is perfectly fine and I can reach the reported perplexity) 

Ubuntu 18.04
pytorch 1.1.0

**Comment by nzmora** [comment]
Hi @hadifar 

Please let us know how you ran the examples - i.e. please provide the full command-line you used and if possible attach also the log file.

Thanks
Neta

**Comment by guyjacob** [comment]
Note that the we did not update the `main.py` script to use `DistillerRNNModel`. Using the Distiller variant + quantization is only shown in the `quantize_lstm.ipynb` notebook. Are you trying to use the notebook, or did you make any changes to `main.py`? If the latter, please specify the changes.

**Comment by hadifar** [comment]
@nzmora @guyjacob Thank you very much for the replies. I update my question. 

I didn't go for quantization. I just simply replace RNNModel with DistillerRNNModel in main.py.

**Comment by hadifar** [comment]
Here is the [log file](https://www.dropbox.com/s/z9o7765ogrt7unh/logs?dl=0) for the first 35 epochs. As can be seen, the ppl is not reaching to ~84/74

https://www.dropbox.com/s/z9o7765ogrt7unh/logs?dl=0

**Comment by levzlotnik** [comment]
Hi @hadifar ,

We didn't test `DistillerLSTM` in the context of training, i.e. our tests do not test the `.backward` of our version of LSTM vs pytorch's `nn.LSTM`.  
If you use `DistillerLSTM` in the context of training, you're relying on the autograd mechanism of pytorch, while `nn.LSTM` was built with CuDNN/MKL-DNN explicit backprop for RNNs - which is probably optimized more than autograd mechanism.  
This may be the main cause of the non-matching results in training.  
I am curious - why would you train a `DistillerLSTM` when you have `nn.LSTM`? 

Thanks,
Lev

**Comment by levzlotnik** [comment]
Hi,
Closing this issue due to staleness. Feel free to reopen if there are further questions.
